### PR TITLE
Drop Swift versions earlier than 5.8

### PR DESCRIPTION
**Pull request by gjcairo**
This PR removes support for Swift versions earlier than 5.8

### Motivation:

Following the changes done in `swift-metrics` on https://github.com/apple/swift-metrics/pull/141, I believe we should align `swift-log` with the rest of the ecosystem too and support only the last 3 Swift releases.

### Modifications:

This PR removes support for Swift versions earlier than 5.8

### Result:

Only the last 3 Swift releases are now supported.

### CODE REVIEW

**Review by ktoso — APPROVED** [review]
Approved but may need to adjust CI right?

**Comment by ktoso** [comment]
@swift-server-bot test this please

**Comment by ktoso** [comment]
Mind checking the soundness (formatting issue)? The CI jobs for old swift we'll remove.

File: scripts/build_podspec.sh [hunk 1]
```diff
@@ -74,7 +74,7 @@ Pod::Spec.new do |s|
   s.documentation_url = 'https://apple.github.io/swift-log'
```
Comment (FranzBusch) [inline scripts/build_podspec.sh:74]: We should delete this file

**Comment by FranzBusch** [comment]
I think the soudness job just needs to be updated to use a newer docker file

**Review by FranzBusch — APPROVED** [review]

File: scripts/build_podspec.sh (hunk 1 above)
Comment (ktoso) [reply scripts/build_podspec.sh:74]: Agreed, kill it

Comment (gjcairo) [reply scripts/build_podspec.sh:74]: Yeah okay I wasn't sure if we were still supporting this or not. FYI: it's still in swift-metrics, so we may want to get rid of it too.

**Comment by gjcairo** [comment]
No, there was an actual whitespace formatting issue - just fixed it.
The pipeline for api breakage has to be updated to use a newer docker image though. I've asked @yim-lee to take care of the CI changes.

**Comment by yim-lee** [comment]
CI updated. @swift-server-bot test this please.

**Comment by gjcairo** [comment]
@swift-server-bot test this please

**Comment by gjcairo** [comment]
Well, I don't think that worked :D @ktoso / @yim-lee mind triggering the tests again?

**Comment by FranzBusch** [comment]
@swift-server-bot test this please

**Comment by FranzBusch** [comment]
@swift-server-bot add to allowlist

**Comment by FranzBusch** [comment]
@gjcairo We gotta update the breaking change checker script the same way we did in metrics.

**Comment by ktoso** [comment]
branch protection rules adjusted in github -- can you fix the api checker?

**Pull request merged**